healpy/fitsfunc.py

```diff
@@ -722,6 +716,8 @@ def getformat(t):
         np.dtype(np.complex64): "C",
         np.dtype(np.complex128): "M",
     }
+    if hasattr(t, "type"):
```

Comment:
This does indeed seem to correctly handle types of specified, non-native endianness. However, this is a very low-level way to go about it. The `astropy.io.fits` module internally has more robust code to map dtypes to FITS TTYPEn cards. I think it would be advisable to rewrite much of this so that it uses higher-level `astropy.io.fits` interfaces and doesn't muck about with dtypes and FITS headers directly.

If `t` is a dtype, then I think that it will always have the `type` attribute. However, if `t` is a string like `>f4`, then the function won't correctly handle it; you would need to first call `np.dtype(t)`.

This function could certainly be simplified to reduce the number of different conversion attempts.

Reply:
it seems `fits.Column` objects still need to manually provide the format.
do you mean to use `Table` objects? https://docs.astropy.org/en/stable/io/unified.html#table-io-fits

Reply:
I agree, it is a large change, targeting next release #699